esp32_s3_lcd_ev_board/include/bsp/esp32_s3_lcd_ev_board.h

```diff
@@ -347,11 +404,41 @@ void bsp_display_unlock(void);
  */
 void bsp_display_rotate(lv_disp_t *disp, lv_disp_rot_t rotation);
 
+/**
+ * @brief Get sub-board type
+ *
+ * @note  This function should be called after calling `bsp_display_new()` or `bsp_display_start()`
+ *
+ * @return
+ *      - SUB_BOARD_TYPE_UNKNOW: Unknow sub-board
+ *      - SUB_BOARD_TYPE_2_480_480: Sub-board 2 with 480x480 LCD (GC9503), Touch (FT5x06)
+ *      - SUB_BOARD_TYPE_3_800_480: Sub-board 3 with 800x480 LCD (ST7262), Touch (GT1151)
+ */
+bsp_sub_board_type_t bsp_display_get_sub_board_type(void);
+
+/**
+ * @brief Get display horizontal resolution
+ *
+ * @note  This function should be called after calling `bsp_display_new()` or `bsp_display_start()`
+ *
+ * @return Horizontal resolution. Return 0 if error occured.
+ */
+uint16_t bsp_display_get_h_res(void);
```

Comment:
Do we need it in public header? (same for `v_res`)

Reply:
Yes, the macros `BSP_LCD_H_RES` and `BSP_LCD_V_RES` are generated by these two functions.